rust/connlib/tunnel/src/client.rs

```diff
@@ -1017,41 +992,9 @@ impl ClientState {
 
             self.resource_ids.remove(id);
 
-            let Some(gateway_id) = self.resources_gateways.remove(id) else {
+            if self.resources_gateways.remove(id).is_none() {
                 tracing::debug!("No gateway associated with resource");
-                continue;
             };
-
-            let Some(peer) = self.peers.get_mut(&gateway_id) else {
-                continue;
-            };
-
-            // First we remove the id from all allowed ips
-            for (network, resources) in peer
-                .allowed_ips
-                .iter_mut()
-                .filter(|(_, resources)| resources.contains(id))
-            {
-                resources.remove(id);
-
-                if !resources.is_empty() {
-                    continue;
-                }
-
-                // If the allowed_ips doesn't correspond to any resource anymore we
-                // clean up any related translation.
-                peer.translations.remove_by_left(&network.network_address());
```

Comment:
ah, I see this is why we keep the `ResourceId` in `allowed_ips`, I think it's a good to keep translation clean but we won't need this anymore after #5049

Reply:
Fair. We can defer merging this until after #5049 is in.